src/beamlime/constructors/providers.py

```diff
@@ -233,10 +233,23 @@ def __init__(
         from functools import lru_cache
 
         super().__init__(_constructor, *args, **kwargs)
+        self._check_args_hashability()
+
         self.cached_result: Product
         self.cached_args = CachedArguments()
         self.cache_indicator = lru_cache(maxsize=2)(self.cached_args)
```

Comment:
What is the reason for the `maxsize=2`? I guess a comment for clarity would help ;-)

Reply:
Yes...!
`CachedProvider` is not supposed to be called with different arguments.
So it is not necessary to have more than 2 cache size... I thought...

Reply:
Wouldn't it be maxsize=1 then?

Reply:
In plopp, we implemented a very simple caching mechanism where we store a variable `self._data` in the nodes that is set to `None` by default.

Could you do something similar where you store the args the first time it is called, and then check if the args are the same on subsequent calls?

This would also remove the annoyances with the hashable args.

Reply:
That is actually what I tried at the beginning and then `==` was not very useful for comparison in many cases, since it automatically falls back to reference comparison, which will be `False` even though they have same `values`.

It's also not clear which method to use for comparison.
When you do `==` between two scipp data arrays or variables, it gives a data array or a variable and then it can't be simply used as a condition of `if` statement.... (right...?) so I had to do sth like `(da == da).all().value` to retrieve `True/False`.

Another reason was that `lru_cache` has a fast way to compare the arguments so I didn't have to deal with that.